search.go

```diff
@@ -1,175 +1,152 @@
-//	results, err := req.Do(context.Background(), client)
+//	searchReq := NewSearchRequest().
+//		WithQuery(NewTermQuery("field", "term"))
+//	searchResp, err := osv2Executor.Search(ctx, searchReq)
+//
+// An error can be returned if
+//   - The request to OpenSearch fails
+//   - The results json cannot be unmarshalled
 type SearchRequest struct {
-	Query search.Query
+	// Query to be performed by the search
+	Query Query
+
+	// Index(s) to be targeted by the search
 	Index []string
-	Size  int
-	Sort  []*search.Sort
+
+	// Size of results to be returned
+	Size int
+
+	// Sort(s) to order the results returned
+	Sort []*Sort
 }
 
 // NewSearchRequest instantiates a SearchRequest with a Size of -1.
 // Any negative value for SearchRequest.Size will be ignored and not included in the source.
 // Opensearch by default, if no size is included in a search request, will limit the results to 10 documents.
 // A NewSearchRequest will search across all indices and return the top 10 documents with the default [sorting].
 //
-// [sorting]: https://opensearch.org/docs/latest/opensearch/search/sort/
+// [sorting]: https://openorg/docs/latest/opensearch/search/sort/
 func NewSearchRequest() *SearchRequest {
 	return &SearchRequest{Size: -1}
 }
 
-// MarshalJSON marshals the SearchRequest into the JSON shape expected by OpenSearch.
-// Implemented to match the [json.Marshaler] interface.
-func (r *SearchRequest) MarshalJSON() ([]byte, error) {
-	source := make(map[string]any)
-	if r.Query != nil {
-		queryJSON, jErr := r.Query.ToOpenSearchJSON()
-		if jErr != nil {
-			return nil, jErr
-		}
-
-		source["query"] = json.RawMessage(queryJSON)
-	}
-
-	if r.Size >= 0 {
-		source["size"] = r.Size
-	}
-
-	if len(r.Sort) > 0 {
-		sorts := make([]json.RawMessage, len(r.Sort))
-		for i, s := range r.Sort {
-			sortJSON, jErr := s.ToOpenSearchJSON()
-			if jErr != nil {
-				return nil, jErr
-			}
-
-			sorts[i] = sortJSON
-		}
-
-		source["sort"] = sorts
-	}
-
-	return json.Marshal(source)
-}
-
-// AddIndices sets the index list for the request.
-func (r *SearchRequest) AddIndices(indices ...string) *SearchRequest {
+// WithIndices sets the index list for the request.
+func (r *SearchRequest) WithIndices(indices ...string) *SearchRequest {
 	r.Index = append(r.Index, indices...)
 	return r
 }
 
-// SetSize sets the request size, limiting the number of documents returned.
+// WithSize sets the request size, limiting the number of documents returned.
 // A negative value for size will be ignored and not included in the SearchRequest.Source.
-func (r *SearchRequest) SetSize(n int) *SearchRequest {
+func (r *SearchRequest) WithSize(n int) *SearchRequest {
 	r.Size = n
 	return r
 }
 
-// AddSort to the current list of [search.Sort]s on the request.
-func (r *SearchRequest) AddSort(sort ...*search.Sort) *SearchRequest {
+// WithSorts to the current list of [Sort]s on the request.
+func (r *SearchRequest) WithSorts(sort ...*Sort) *SearchRequest {
```

Comment:
Same here.  We should consider renaming this or have 2 versions: `WithSorts()` does a replacement and `AddSorts()`/`AppendSorts()` does what this method is doing.